Taskfile.yml

```diff
@@ -0,0 +1,109 @@
+# yaml-language-server: $schema: https://taskfile.dev/schema.json
+
+# Install `task` by following https://taskfile.dev/installation/
+# Run `task up` to start argo.
+
+# - `env PROFILE=mysql task up` to start with MySQL.
+# - `env PROFILE=plugins ARGO_EXECUTOR_PLUGINS=true task up` to start with plugins.
+# - `env PROFILE=sso ARGO_AUTH_MODE=sso task up` to start with SSO.
+
+# The app will be up-and-running between 15s and 1m later (if hot compiled or cold).
+# When using `--watch`, any changes made to the source code will be automatically recompiled and the app restarted, typically within a few seconds.
+
+version: 3
+
+tasks:
+  go-deps:
+    cmd: go mod download
+    # mutex: download
```

Comment:
I don't think any of these mutexes are actually strictly necessary, two downloads and two builds can run in parallel, they just may bottleneck, so it might be a concurrency optimization to run them serially instead.

Task does have the [`run`](https://taskfile.dev/usage/#limiting-when-tasks-run) key to ensure something runs only once. And it does have a [serial syntax](https://taskfile.dev/usage/#calling-another-task), but it's not as flexible as a string for a mutex lock.

Reply:
Removed the `download` mutex in #13350